finetuner/tuner/pytorch/miner.py

```diff
@@ -34,11 +49,61 @@ def mine(
         :return: triplet of label indices follows the order of anchor, positive and negative.
         """
         assert len(embeddings) == len(labels)
-        labels1 = np.expand_dims(labels, 1)
-        labels2 = np.expand_dims(labels, 0)
-        matches = (labels1 == labels2).astype(int)
-        diffs = matches ^ 1
-        np.fill_diagonal(matches, 0)
-        triplets = np.expand_dims(matches, 2) * np.expand_dims(diffs, 1)
-        idxes_anchor, idxes_pos, idxes_neg = np.where(triplets)
-        return list(zip(idxes_anchor, idxes_pos, idxes_neg))
+        return _generate_all_possible_triplets(labels)
+
+
+class SiameseSessionMiner(BaseMiner):
+    def mine(
+        self, embeddings: List[AnyTensor], labels: List[Tuple[int, int]]
```

Comment:
```suggestion
        self, embeddings: AnyTensor, labels: List[Tuple[int, int]]
```

It's not a list, simply a tensor with first dimension `N`